libs/dyn/dynvar/ref.go

```diff
@@ -6,7 +6,7 @@ import (
 	"github.com/databricks/cli/libs/dyn"
 )
 
-const VariableRegex = `\$\{([a-zA-Z]+([-_]?[a-zA-Z0-9]+)*(\.[a-zA-Z]+([-_]?[a-zA-Z0-9]+)*)*)\}`
+const VariableRegex = `\$\{([a-zA-Z]+([-_]?[a-zA-Z0-9]+)*(\.[a-zA-Z]+([-_]?[a-zA-Z0-9]+)*(\[[0-9]+\])?)*(\[[0-9]+\])?)\}`
```

Comment:
This supports `var.foo[0]`, but not `var.foo[0][1]`.

Reply:
Good catch, forgot that it can be nested